### ISSUE TITLE
Python bindings build does not respect compiler settings

### ISSUE THREAD

**Issue by cjones051073** [xrootd/xrootd#848]
It seems the build of the python bindings at the end of the build job do not respect the compiler selection made for the main library. For instance I have a build configured with

```
Executing:  cd "/opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/build" && /opt/local/bin/cmake -DCMAKE_INSTALL_PREFIX='/opt/local' -DCMAKE_BUILD_TYPE=Release -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON -DCMAKE_C_COMPILER="$CC" -DCMAKE_COLOR_MAKEFILE=ON -DCMAKE_CXX_COMPILER="$CXX" -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_INSTALL_NAME_DIR=/opt/local/lib -DCMAKE_INSTALL_RPATH=/opt/local/lib -DCMAKE_MODULE_PATH=/opt/local/share/cmake/Modules -DCMAKE_SYSTEM_PREFIX_PATH="/opt/local;/opt/local;/usr" -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -Wno-dev -DENABLE_FUSE=False -DCMAKE_INSTALL_MANDIR="share/man" -DENABLE_READLINE=True -DREADLINE_ROOT_DIR="/opt/local" -DENABLE_CRYPTO=True -DOPENSSL_ROOT_DIR="/opt/local" -DENABLE_KRB5=True -DKERBEROS5_ROOT_DIR="/opt/local" -DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_OSX_ARCHITECTURES="x86_64" -DCMAKE_OSX_DEPLOYMENT_TARGET="10.10" -DCMAKE_OSX_SYSROOT="/" /opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/xrootd-4.8.5 
-- The C compiler identification is Clang 6.0.1
-- The CXX compiler identification is Clang 6.0.1
-- Check for working C compiler: /opt/local/bin/clang-mp-6.0
-- Check for working C compiler: /opt/local/bin/clang-mp-6.0 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/local/bin/clang++-mp-6.0
-- Check for working CXX compiler: /opt/local/bin/clang++-mp-6.0 -- works
<snip>
```
this selection is correctly used for the main build.

```
[  2%] Building CXX object src/CMakeFiles/XrdCnsLib.dir/XrdCns/XrdCnsLogRec.cc.o
cd /opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/build/src && /opt/local/bin/clang++-mp-6.0  -DHAVE_ATOMICS -DHAVE_CRYPT -DHAVE_ET_COM_ERR_H -DHAVE_FSTATAT -DHAVE_GETIFADDRS -DHAVE_LIBZ -DHAVE_NAMEINFO -DHAVE_READLINE -DHAVE_SSL -DHAVE_STRLCPY -DHAVE_TLS1 -DHAVE_TLS11 -DHAVE_TLS12 -DHAVE_XML2 -DHAVE_XRDCRYPTO -DLT_MODULE_EXT=\".dylib\" -DUSE_LIBC_SEMAPHORE=0 -DXRDPLUGIN_SOVERSION=\"4\" -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D__macos__=1 -I/opt/local/include -I/opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/xrootd-4.8.5/src/.. -I/opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/xrootd-4.8.5/src/. -I/opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/xrootd-4.8.5/src -I/opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/build/src -I/usr/local/include  -pipe -Os -stdlib=libc++ -std=c++0x -Wno-deprecated-declarations -DNDEBUG -arch x86_64 -mmacosx-version-min=10.10   -o CMakeFiles/XrdCnsLib.dir/XrdCns/XrdCnsLogRec.cc.o -c /opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/xrootd-4.8.5/src/XrdCns/XrdCnsLogRec.cc
```

but then later on is ignored when the bindings are built.

```
creating build/temp.macosx-10.10-x86_64-2.7/opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/xrootd-4.8.5/bindings/python/src
/usr/bin/clang -fno-strict-aliasing -fno-common -dynamic -pipe -Os -DNDEBUG -g -fwrapv -O3 -Wall -I/opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/xrootd-4.8.5/src -I/opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/build/src -I/opt/local/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c /opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/xrootd-4.8.5/bindings/python/src/PyXRootDCopyProcess.cc -o build/temp.macosx-10.10-x86_64-2.7/opt/local/var/macports/build/_Users_chris_Projects_MacPorts_ports_science_xrootd/xrootd/work/xrootd-4.8.5/bindings/python/src/PyXRootDCopyProcess.o -g
```

see the use of `/usr/bin/clang` instead of the configured compiler.

My expectation is the configured compiler should be used for both.

**Comment by alexpearce** [comment]
Could #695 be a fix for this?

**Comment by simonmichal** [comment]
Yep, this has been fixed in #695.
